### DISCUSSION TITLE
If the existing app has R6 classes in logic (e.g. for database connection) - is it more sensible to re-implement using a box/modules approach?

### DISCUSSION THREAD

**Discussion by FMKerckhof** [Appsilon/rhino#291] [Q&A] [answered]
With Rhino, if you want to migrate an existing app that uses `R6class` in the logic (e.g. for a database connection) - is it more sensible to rather create the public functions and `#' export` each of them for use with `box::use`?

**Comment by kamilzyla** [comment, marked as answer]
I'd say box and R6 are perpendicular to each other. They serve a somewhat different purpose: R6 classes let you bundle data and code which operates on it; box modules let you structure the code of your application and express dependencies clearly. Feel free to use box and R6 together - just mark your R6 class with `#' @export` and import it where you need it.

**Reply by FMKerckhof** [reply]
thanks Kamil, this is very clear. I got a bit confused on the modularization with the exports of the R6 public functions. But now I understand I should just export the entire class.
